app/src/main/java/io/gnosis/safe/ui/safe/settings/safe/SafeSettingsViewModel.kt

```diff
@@ -0,0 +1,50 @@
+package io.gnosis.safe.ui.safe.settings.safe
+
+import io.gnosis.data.repositories.SafeRepository
+import io.gnosis.safe.Tracker
+import io.gnosis.safe.ui.base.AppDispatchers
+import io.gnosis.safe.ui.base.BaseStateViewModel
+import kotlinx.coroutines.flow.collect
+import javax.inject.Inject
+
+class SafeSettingsViewModel @Inject constructor(
+    private val safeRepository: SafeRepository,
+    private val tracker: Tracker,
+    appDispatchers: AppDispatchers
+) : BaseStateViewModel<SafeSettingsState>(appDispatchers) {
+
+    override fun initialState() = SafeSettingsState(ViewAction.Loading(true))
+
+    init {
+        safeLaunch {
+            safeRepository.activeSafeFlow().collect { safe ->
+                updateState(true) {
+                    SafeSettingsState(ViewAction.UpdateActiveSafe(safe))
+                }
+            }
+        }
+    }
+
+    fun removeSafe() {
+        safeLaunch {
+            runCatching {
+                val safe = safeRepository.getActiveSafe()
+                safe?.let {
+                    safeRepository.removeSafe(safe)
+                    safeRepository.setActiveSafe(null)
```

Comment:
maybe this line should be part of the `removeSafe` method? We should encapsulate the whole logic of removing safe within the repository, in case in the future we should be able to remove safes from somewhere else in the app.

Reply:
I would like to have these operations as atomic. This is the cleaner approach I think. How they are combined should be decided by our business logic